execute/plugin_e2e_test.go

```diff
@@ -188,3 +188,79 @@ func TestPlugin_FinalizedUnfinalizedCaching(t *testing.T) {
 	report := outcome.CommitReports[0]
 	require.Equal(t, cciptypes.NewSeqNumRange(200, 201), report.SequenceNumberRange)
 }
+
+func TestPlugin_CommitReportTimestampOrdering(t *testing.T) {
+	ctx := tests.Context(t)
+
+	srcChainA := cciptypes.ChainSelector(1)
+	dstSelector := cciptypes.ChainSelector(2)
+
+	// Create messages for multiple commit reports with different timestamps
+	baseTime := time.Now().Add(-4 * time.Hour)
+	msgSets := []struct {
+		messages  []inmem.MessagesWithMetadata
+		timestamp time.Time
+		blockNum  uint64
+	}{
+		{
+			// Latest messages
+			messages: []inmem.MessagesWithMetadata{
+				makeMsgWithMetadata(100, srcChainA, dstSelector, false),
+				makeMsgWithMetadata(101, srcChainA, dstSelector, false),
+			},
+			timestamp: baseTime.Add(2 * time.Hour),
+			blockNum:  1002,
+		},
+		{
+			messages: []inmem.MessagesWithMetadata{
+				makeMsgWithMetadata(102, srcChainA, dstSelector, false),
+				makeMsgWithMetadata(103, srcChainA, dstSelector, false),
```

Comment:
This is a nit but the contents of this test data doesn't match with how CCIP works - older reports should have smaller sequence numbers.

Maybe:

```suggestion
				makeMsgWithMetadata(102, srcChainA, dstSelector, false),
				makeMsgWithMetadata(103, srcChainA, dstSelector, false),
			},
			timestamp: baseTime.Add(2 * time.Hour), // more recent
			blockNum:  1002,
		},
		{
			messages: []inmem.MessagesWithMetadata{
				makeMsgWithMetadata(100, srcChainA, dstSelector, false),
				makeMsgWithMetadata(101, srcChainA, dstSelector, false),
```